crates/router/src/core/payments/operations/payment_create.rs

```diff
@@ -1023,6 +1032,69 @@ impl PaymentCreate {
             .change_context(errors::ApiErrorResponse::InternalServerError)?
             .map(Secret::new);
 
+        let guest_customer_details_json = if let Some(cid) = &request.customer_id {
+            let customer = state
+                .store
+                .find_customer_by_customer_id_merchant_id(
+                    cid,
+                    &merchant_account.merchant_id,
+                    key_store,
+                    merchant_account.storage_scheme,
+                )
+                .await
+                .to_not_found_response(errors::ApiErrorResponse::CustomerNotFound);
```

Comment:
we can try to avoid this database call. This function can only store the details for guest customers ( the case where `customer_id` is not present ). In other cases ( cases where customer_id ) is present this can be done in `update_trackers`, where we would have already fetched the customer details